delfin/drivers/dell_emc/vmax/client.py

```diff
@@ -287,6 +287,124 @@ def list_volumes(self, storage_id):
             LOG.error(msg)
             raise exception.StorageBackendException(msg)
 
+    def list_controllers(self, storage_id):
+        try:
+            # Get list of Directors
+            directors = self.rest.get_director_list(self.array_id,
+                                                    self.uni_version)
+            controller_list = []
+            for director in directors:
+                director_info = self.rest.get_director(
+                    self.array_id, self.uni_version, director)
+
+                status = constants.ControllerStatus.NORMAL
+                if director_info.get('availability', '').upper() != 'ONLINE':
+                    status = constants.ControllerStatus.OFFLINE
+
+                controller = {
+                    'name': director_info['directorId'],
+                    'storage_id': storage_id,
+                    'native_controller_id': director_info['directorId'],
+                    'status': status,
+                    'location':
+                        'slot_' +
+                        str(director_info.get('director_slot_number')),
+                    'soft_version': None,
+                    'cpu_info': 'Cores-'
+                                + str(director_info.get('num_of_cores')),
+                    'memory_size': None
+
+                }
+                controller_list.append(controller)
+            return controller_list
+
+        except exception.SSLCertificateFailed:
+            LOG.error('SSL certificate failed when list pools for VMax')
+            raise
+        except Exception as err:
+            msg = "Failed to get controller metrics from VMAX: {}".format(err)
+            LOG.error(msg)
+            raise exception.ControllerNotFound(self.array_id)
```

Comment:
Is it correct to wrap all the exception except SSLCertificateFailed to ControllerNotFound?
For example, if user modify the password on the device but not modify the access info in delfin, it would throw ControllerNotFound here.
Same as other list API.

Reply:
Thanks, updated to raise generated exception.